### PR TITLE
[Modules] Updated Consumption budgets to new dependencies approach

### PR DESCRIPTION
**Pull request by MrMCake**
# Description

- Updated Consumption budgets to new dependencies approach

| Pipeline |
| - |
| [Consumption: Budgets](https://github.com/Azure/ResourceModules/actions/workflows/ms.consumption.budgets.yml) |

# Type of Change

Please delete options that are not relevant.

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

### CODE REVIEW

File: modules/Microsoft.Consumption/budgets/.test/default/deploy.test.bicep [hunk 1]
```diff
@@ -0,0 +1,29 @@
+targetScope = 'subscription'
+
+// ========== //
+// Parameters //
+// ========== //
+@description('Optional. A short identifier for the kind of deployment .Should be kept short to not run into resource-name length-constraints')
```
Comment (eriqua) [inline modules/Microsoft.Consumption/budgets/.test/default/deploy.test.bicep:6]: Minor typo in this description. Applies to all PRs. Should probably be:
```suggestion
@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
```

File: modules/Microsoft.Consumption/budgets/deploy.bicep [hunk 2]
```diff
@@ -40,13 +40,13 @@ param thresholds array = [
   110
 ]
 
-@description('Optional. The list of email addresses to send the budget notification to when the thresholds are exceeded.')
+@description('Conditional. The list of email addresses to send the budget notification to when the thresholds are exceeded. Required if neither `contactRoles` or `actionGroups` was provided.')
```
Comment (eriqua) [inline modules/Microsoft.Consumption/budgets/deploy.bicep:43]:
```suggestion
@description('Conditional. The list of email addresses to send the budget notification to when the thresholds are exceeded. Required if neither `contactRoles` nor `actionGroups` was provided.')
```

File: modules/Microsoft.Consumption/budgets/deploy.bicep [hunk 3]
```diff
@@ -40,13 +40,13 @@ param thresholds array = [
...
 param contactEmails array = []
 
-@description('Optional. The list of contact roles to send the budget notification to when the thresholds are exceeded.')
+@description('Conditional. The list of contact roles to send the budget notification to when the thresholds are exceeded. Required if neither `contactEmails` or `actionGroups` was provided.')
```
Comment (eriqua) [inline modules/Microsoft.Consumption/budgets/deploy.bicep:46]:
```suggestion
@description('Conditional. The list of contact roles to send the budget notification to when the thresholds are exceeded. Required if neither `contactEmails` nor `actionGroups` was provided.')
```

File: modules/Microsoft.Consumption/budgets/deploy.bicep [hunk 4]
```diff
@@ -40,13 +40,13 @@ param thresholds array = [
...
 param contactRoles array = []
 
-@description('Optional. List of action group resource IDs that will receive the alert.')
+@description('Conditional. List of action group resource IDs that will receive the alert. Required if neither `contactEmails` or `contactEmails` was provided.')
```
Comment (eriqua) [inline modules/Microsoft.Consumption/budgets/deploy.bicep:49]:
```suggestion
@description('Conditional. List of action group resource IDs that will receive the alert. Required if neither `contactEmails` nor `contactEmails` was provided.')
```

File: modules/Microsoft.Consumption/budgets/readme.md [hunk 5]
```diff
@@ -24,13 +24,17 @@ This module deploys budgets for subscriptions.
 | `amount` | int | The total amount of cost or usage to track with the budget. |
 | `name` | string | The name of the budget. |
 
+**Conditional parameters**
+| Parameter Name | Type | Description |
+| :-- | :-- | :-- |
+| `actionGroups` | array | List of action group resource IDs that will receive the alert. Required if neither `contactEmails` or `contactEmails` was provided. |
+| `contactEmails` | array | The list of email addresses to send the budget notification to when the thresholds are exceeded. Required if neither `contactRoles` or `actionGroups` was provided. |
+| `contactRoles` | array | The list of contact roles to send the budget notification to when the thresholds are exceeded. Required if neither `contactEmails` or `actionGroups` was provided. |
```
Comment (eriqua) [inline modules/Microsoft.Consumption/budgets/readme.md:32]:
```suggestion
| `actionGroups` | array | List of action group resource IDs that will receive the alert. Required if neither `contactEmails` nor `contactEmails` was provided. |
| `contactEmails` | array | The list of email addresses to send the budget notification to when the thresholds are exceeded. Required if neither `contactRoles` nor `actionGroups` was provided. |
| `contactRoles` | array | The list of contact roles to send the budget notification to when the thresholds are exceeded. Required if neither `contactEmails` nor `actionGroups` was provided. |
```

File: modules/Microsoft.Consumption/budgets/.test/default/deploy.test.bicep [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+param serviceShort string = 'cbdef'
```
Comment (eriqua) [inline modules/Microsoft.Consumption/budgets/.test/default/deploy.test.bicep:7]: All `def `should probably be `com` now?
```suggestion
param serviceShort string = 'cbcom'
```

File: modules/Microsoft.Consumption/budgets/.test/default/deploy.test.bicep (hunk 1 above)
Comment (MrMCake) [reply modules/Microsoft.Consumption/budgets/.test/default/deploy.test.bicep:6]: Updated via automation :)

File: modules/Microsoft.Consumption/budgets/.test/default/deploy.test.bicep (hunk 6 above)
Comment (MrMCake) [reply modules/Microsoft.Consumption/budgets/.test/default/deploy.test.bicep:7]: Updated via automation :)

**Review by eriqua — APPROVED** [review]
🆗

**Pull request merged**